src/modules/dashboard/components/ExpenditurePage/Stages/ClaimFunds/ClaimFunds.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useMemo } from 'react';
+import {
+  defineMessages,
+  FormattedMessage,
+  MessageDescriptor,
+  useIntl,
+} from 'react-intl';
+import classNames from 'classnames';
+import { nanoid } from 'nanoid';
+
+import { FormSection } from '~core/Fields';
+import Tag from '~core/Tag';
+import TimeRelativeShort from '~dashboard/ExpenditurePage/TimeRelativeShort/TimeRelativeShort';
+import Button from '~core/Button';
+import Numeral from '~core/Numeral';
+import { getTokenDecimalsWithFallback } from '~utils/tokens';
+
+import { buttonStyles } from '../Stages';
+import styles from './ClaimFunds.css';
+import { Recipient } from '~dashboard/ExpenditurePage/hooks';
+
+const displayName = 'dashboard.ExpenditurePage.ClaimFunds';
+
+const MSG = defineMessages({
+  claim: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.claim',
+    defaultMessage: 'Next claim {claimDate}',
+  },
+  claimFunds: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.claimFunds',
+    defaultMessage: 'Claim funds',
+  },
+  totalClaimable: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.totalClaimable',
+    defaultMessage: 'Total claimable',
+  },
+  claimableNow: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.claimableNow',
+    defaultMessage: 'Claimable now',
+  },
+  claimed: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.claimed',
+    defaultMessage: 'Claimed',
+  },
+  nothingToClaim: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.nothingToClaim',
+    defaultMessage: 'Nothing to claim',
+  },
+  now: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.now',
+    defaultMessage: 'now',
+  },
+});
+
+export type Token = Record<string, number>;
+
+interface TokenWithId {
```

Comment:
We have a type for token `Token` or `AnyToken`. You can remove any fields that wouldn't be used. I made a log and I see all of this for token:
```
{
token:
address: "0x47b3CfC8EeaEae8Ad83845a19112BeA8ca961fb8"
decimals: 18
iconHash: null
id: "0x47b3CfC8EeaEae8Ad83845a19112BeA8ca961fb8"
name: "b"
processedBalances: (4) [{…}, {…}, {…}, {…}]
symbol: "B"
__typename: "Token"
}
```